source/source.go

```diff
@@ -51,99 +49,29 @@ type Source struct {
 	iterator Iterator
 }
 
+func (s *Source) Config() sdk.SourceConfig {
+	return &s.config
+}
+
 // NewSource creates a new instance of the [Source].
 func NewSource() sdk.Source {
 	return sdk.SourceWithMiddleware(
-		&Source{},
-		sdk.DefaultSourceMiddleware(
-			// disable schema extraction by default, because the source produces raw data
-			sdk.SourceWithSchemaExtractionConfig{
-				PayloadEnabled: lang.Ptr(false),
-				KeyEnabled:     lang.Ptr(false),
+		&Source{
+			config: Config{
+				DefaultSourceMiddleware: sdk.DefaultSourceMiddleware{
+					SourceWithSchemaExtraction: sdk.SourceWithSchemaExtraction{
+						PayloadEnabled: lang.Ptr(false),
+						KeyEnabled:     lang.Ptr(false),
```

Comment:
Acceptance tests were failing with structured data and schema extraction enabled, so I disabled it. I'll open an issue for this so we can handle it separately.

Reply:
https://github.com/conduitio-labs/conduit-connector-mongo/issues/154